syntax = "proto3";

package example;

service RsyncService {
 rpc RsyncMethod (ClientRequest) returns (ServerResponse) {};
}

message ClientRequest {
 string filename = 1;  
 float timestamp = 2; 
};

message ServerResponse {
 repeated File files = 1;
 float timestamp = 2;
};

message File {
 string filename = 1;
 bool isDirectory = 2;
 bytes fileContent = 3;
}

